src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/taglets/TagletWriter.java

```diff
@@ -370,7 +372,8 @@ public Content createAnchorAndSearchIndex(Element element, String tagText, Strin
     @SuppressWarnings("preview")
     Content createAnchorAndSearchIndex(Element element, String tagText, Content tagContent, String desc, DocTree tree) {
         Content result;
-        if (context.isFirstSentence && context.inSummary || context.inTags.contains(DocTree.Kind.INDEX)) {
+        if (context.isFirstSentence && context.inSummary || context.inTags.contains(DocTree.Kind.INDEX)
```

Comment:
If you introduce `context.isIndexable` then the code may read better if you flip the then/else parts.

```
if (htmlWriter.isIndexable() && context.isIndexable()) {
    // index it
} else {
    // don't index it
}
```

Reply:
Let's leave it out for now. It's not because I don't like flipping the `if` and `else` arms around, it's because I'm not very comfortable with introducing `isIndexable` to Context at the moment.